test/GitcoinGovernor.t.sol

```diff
@@ -1086,3 +1081,255 @@ contract NewGitcoinGovernorProposalTest is GitcoinGovernorProposalTestHelper {
     governorAlpha.queue(_vars.alphaProposalId);
   }
 }
+
+// forgefmt: disable-start
+// Copied from Solmate
+library FixedPointMathLib {
+  uint256 internal constant MAX_UINT256 = 2**256 - 1;
+
+  uint256 internal constant WAD = 1e18; // The scalar of ETH and most ERC20s.
+
+  function mulWadDown(uint256 x, uint256 y) internal pure returns (uint256) {
+    return mulDivDown(x, y, WAD); // Equivalent to (x * y) / WAD rounded down.
+  }
+
+  function mulDivDown(
+    uint256 x,
+    uint256 y,
+    uint256 denominator
+  ) internal pure returns (uint256 z) {
+    assembly {
+      // Equivalent to require(denominator != 0 && (y == 0 || x <= type(uint256).max / y))
+      if iszero(mul(denominator, iszero(mul(y, gt(x, div(MAX_UINT256, y)))))) {
+        revert(0, 0)
+      }
+
+      // Divide x * y by the denominator.
+      z := div(mul(x, y), denominator)
+    }
+  }
+}
+// forgefmt: disable-end
+
+contract FlexVoting is GitcoinGovernorProposalTestHelper, GovernorBravoProposalHelper {
+  using FixedPointMathLib for uint256;
+
+  // Store the id of a new proposal unrelated to governor upgrade.
+  uint256 newProposalId;
+
+  event VoteCastWithParams(
+    address indexed voter,
+    uint256 proposalId,
+    uint8 support,
+    uint256 weight,
+    string reason,
+    bytes params
+  );
+
+  function setUp()
+    public
+    virtual
+    override(GitcoinGovernorProposalTestHelper, GitcoinGovernorTestHelper)
+  {
+    GitcoinGovernorProposalTestHelper.setUp();
+
+    _upgradeToBravoGovernor();
+
+    (newProposalId,,,,) = _submitTokenSendProposalToGovernorBravo(
+      address(usdcToken), usdcToken.balanceOf(TIMELOCK), makeAddr("receiver for FlexVoting tests")
+    );
+
+    _jumpToActiveProposal(newProposalId);
+  }
+
+  function testFuzz_GovernorBravoSupportsCastingSplitVotes(
```

Comment:
Looks like these tests aren't copy/pasta'd from the Fractional repo because they are testing the Governor in situ after the upgrade against the fork. In other words, it's testing that fractional voting works post upgrade, with the GitcoinGovernor assembled according to Gitcoin's requirements, which is not what we do in the Fractional Governor unit tests. In fact that's the point of all these tests, as the Governor components themselves are in fact tested by OZ.

Reply:
> In other words, it's testing that fractional voting works post upgrade, with the GitcoinGovernor assembled according to Gitcoin's requirements, which is not what we do in the Fractional Governor unit tests.

So IIUC we're basically testing that the fractional voting logic wasn't overridden accidentally when assembling the governor?

Reply:
> What's the testing philosophy you took here?

Good question. I wrote new tests to cover the flex voting features that I thought Gitcoin would care about having, as well as some sanity tests that those features wouldn't interfere with normal governor functioning. Otherwise I didn't retest `GovernorCountingFractional`, since we've already done that

Reply:
Thanks, appreciate the explanation!